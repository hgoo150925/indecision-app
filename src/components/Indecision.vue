<template>
  <img v-if="image" :src="image" alt="bg" />
  <div class="bg-dark"></div>
  <div class="indecision-container">
    <input v-model="question" type="text" placeholder="Hazme una pregunta" />
    <p>Recuerda terminar con un signo de interrogacion</p>
    <!-- isValidQuestion si la pregunta es valida, se muestra la pregunta -->
    <div v-if="isValidQuestion">
      <h2>{{ question }}</h2>
      <h1>{{ answer }}</h1>
    </div>
  </div>
</template>

<script>
export default {
  data() {
    return {
      question: '',
      answer: null,
      image: null,
      isValidQuestion: false,
    };
  },

  // https://vuejs.org/guide/essentials/watchers.html#basic-example
  // Con watch puedo estar pendiente a los cambios en la propiedad reactiva question
  // Si el valor de question cambia, puedo realizar una accion
  // Basicamente watch lo que hace es "observar" si hay un cambio o no en el estado de la propiedad, en este caso question
  // Si existe ese cambio, watch permite realizar cualquier accion que querramos
  watch: {
    question(newValue, oldValue) {
      this.isValidQuestion = false;
      // Si el nuevo valor de question no incluye un singo de interrogacion, que no haga nada
      if (!newValue.includes('?')) {
        return;
      }
      this.isValidQuestion = true;
      // https://yesno.wtf/
      this.getAnswer();
    },
  },

  methods: {
    async getAnswer() {
      this.answer = 'Pensando...';
      try {
        const { answer, image } = await fetch('https://yesno.wtf/api').then(
          res => res.json()
        );
        this.answer = answer === 'yes' ? 'Si' : 'No';
        this.image = image;
      } catch (error) {
        this.answer = `Error! Could not reach the API. ${error}`;
      }
    },
  },
};
</script>

<style scoped>
img,
.bg-dark {
  height: 100vh;
  left: 0px;
  max-height: 100%;
  max-width: 100%;
  position: fixed;
  top: 0px;
  width: 100vw;
}

.bg-dark {
  background-color: rgba(0, 0, 0, 0.4);
}

.indecision-container {
  position: relative;
  z-index: 99;
}

input {
  width: 250px;
  padding: 10px 15px;
  border-radius: 5px;
  border: none;
}
input:focus {
  outline: none;
}

p {
  color: white;
  font-size: 20px;
  margin-top: 0px;
}

h1,
h2 {
  color: white;
}

h2 {
  margin-top: 150px;
}
</style>
