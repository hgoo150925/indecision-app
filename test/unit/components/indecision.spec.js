import { describe, test, expect } from 'vitest'
import { shallowMount } from '@vue/test-utils'

import Indecision from '../../../src/components/Indecision.vue'


describe('Indecision component', () => {
    test('debe de ser identico con el snapshot', () => {
        const wrapper = shallowMount(Indecision)
        // toMatchSnapshot() garantiza que un valor coincida con la instantánea más reciente.
        expect(wrapper.html()).toMatchSnapshot()

        // actualizar snapshot
        // yarn test -u
    })
})