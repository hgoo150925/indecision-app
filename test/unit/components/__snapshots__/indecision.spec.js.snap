// Vitest Snapshot v1

exports[`Indecision component > debe de ser identico con el snapshot 1`] = `
"<!--v-if-->
<div class=\\"bg-dark\\" data-v-4a8fca50=\\"\\"></div>
<div class=\\"indecision-container\\" data-v-4a8fca50=\\"\\"><input type=\\"text\\" placeholder=\\"Hazme una pregunta\\" data-v-4a8fca50=\\"\\">
  <p data-v-4a8fca50=\\"\\">Recuerda terminar con un signo de interrogacion</p><!-- isValidQuestion si la pregunta es valida, se muestra la pregunta -->
  <!--v-if-->
</div>"
`;
